### PR TITLE
Feat/DEVEXP-470: Voice V1 design

### PR DESCRIPTION
**Pull request by JPPortier**

### CODE REVIEW

File: client/src/main/com/sinch/sdk/domains/voice/api/v1/ApplicationsService.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+   */
+  void assignNumbers(UpdateNumbersRequest parameters);
+
+  /**
+   * Un-assign a number from an application.
+   *
+   * @param parameters Request parameters
+   * @since 1.4
+   */
+  public void unAssignNumber(UnAssignNumberRequest parameters);
```
Comment (asein-sinch) [inline client/src/main/com/sinch/sdk/domains/voice/api/v1/ApplicationsService.java]: Nit: `unAssign` or `unassign`, that is the question 💀

File: client/src/main/com/sinch/sdk/domains/voice/api/v1/adapters/WebHooksService.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      String method, String path, Map<String, String> headers, String jsonPayload) {
+
+    // convert header keys to use case-insensitive map keys
+    Map<String, String> caseInsensitiveHeaders = MapUtils.getCaseInsensitiveMap(headers);
+
+    String authorizationHeader = caseInsensitiveHeaders.get("Authorization");
+
+    // no authorization required
+    if (null == authorizationHeader) {
+      return true;
```
Comment (asein-sinch) [inline client/src/main/com/sinch/sdk/domains/voice/api/v1/adapters/WebHooksService.java]: Shouldn't it be `false` if no authorization header is present?

File: client/src/test/java/com/sinch/sdk/domains/voice/adapters/CallsServiceTest.java [hunk 3]
```diff
@@ -30,67 +26,58 @@
...
-    SVAMLRequestBodyDto body = updateParametersCaptor.getValue();
-    Assertions.assertThat(body).isEqualTo(CallsDtoConverterTest.svamlRequestBodyDto);
+    SvamlControl body = updateParametersCaptor.getValue();
+    TestHelpers.recursiveEquals(body, CallsDtoConverterTest.svamlControlDto);
   }
 
   @Test
   void manageWithCallLeg() throws ApiException {
 
     service.manageWithCallLeg("call id", CallLegType.BOTH, SVAMLControlTest.parameters);
```
Comment (asein-sinch) [inline client/src/test/java/com/sinch/sdk/domains/voice/adapters/CallsServiceTest.java:68]: Should it be `CallLegType` or `CallLeg` (from v1) ?

File: client/src/main/com/sinch/sdk/domains/voice/api/v1/ApplicationsService.java (hunk 1 above)
Comment (JPPortier) [reply client/src/main/com/sinch/sdk/domains/voice/api/v1/ApplicationsService.java]: Exact and was not confirmed to our mappingId -> name mapping table
Fixed

File: client/src/main/com/sinch/sdk/domains/voice/api/v1/adapters/WebHooksService.java (hunk 2 above)
Comment (JPPortier) [reply client/src/main/com/sinch/sdk/domains/voice/api/v1/adapters/WebHooksService.java]: Indeed !
Fixed
Side note: not addressed with this PR, but a ticket is existing in queue to extend unit tests related to Voice/Verification about authentication (DEVEXP-404)

File: client/src/test/java/com/sinch/sdk/domains/voice/adapters/CallsServiceTest.java (hunk 3 above)
Comment (JPPortier) [reply client/src/test/java/com/sinch/sdk/domains/voice/adapters/CallsServiceTest.java:68]: Not here: we are testing "old" design service still working: https://developers.sinch.com/java-sdk/apidocs/com/sinch/sdk/domains/voice/CallsService.html#manageWithCallLeg(java.lang.String,com.sinch.sdk.domains.voice.models.CallLegType,com.sinch.sdk.domains.voice.models.svaml.SVAMLControl)
So it accept a `CallLegType` and should be wrapped internally onto "new design" service based onto CallLeg.
This is why :
- https://github.com/sinch/sinch-sdk-java/pull/164/files#diff-be39bcdae0377a57e69f1a79400a0628c221f8c3859acfd3fd970eb23c0e7ef7R32
- and https://github.com/sinch/sinch-sdk-java/pull/164/files#diff-be39bcdae0377a57e69f1a79400a0628c221f8c3859acfd3fd970eb23c0e7ef7R77

We are validating old design CallLegtype was correctly transformed to new `CallLeg` (type and value)

File: client/src/test/java/com/sinch/sdk/domains/voice/adapters/CalloutsServiceTest.java [hunk 4]
```diff
@@ -67,8 +58,20 @@ void callTTS() throws ApiException {
   @Test
   void callCustom() throws ApiException {
```
Comment (asein-sinch) [inline client/src/test/java/com/sinch/sdk/domains/voice/adapters/CalloutsServiceTest.java]: Nit: to be renamed `custom()` to be aligned with the name of the tested method as it has been done for `conference()` and `textToSpeech()`

File: client/src/test/java/com/sinch/sdk/domains/voice/adapters/CallsServiceTest.java (hunk 3 above)
Comment (asein-sinch) [reply client/src/test/java/com/sinch/sdk/domains/voice/adapters/CallsServiceTest.java:68]: Thanks for your explanation!

File: client/src/test/java/com/sinch/sdk/domains/voice/api/v1/adapters/VoiceServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.sinch.sdk.core.http.HttpClient;
+import com.sinch.sdk.models.ApplicationCredentials;
+import com.sinch.sdk.models.VoiceContext;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+class VoiceServiceTest {
```
Comment (asein-sinch) [inline client/src/test/java/com/sinch/sdk/domains/voice/api/v1/adapters/VoiceServiceTest.java:12]: Can you also add a test where there is a context, an application key and an application secret? Just to be sure it works when all the parameters are present :)

File: client/src/main/com/sinch/sdk/domains/voice/api/v1/ApplicationsService.java (hunk 1 above)
Comment (asein-sinch) [reply client/src/main/com/sinch/sdk/domains/voice/api/v1/ApplicationsService.java]: Thanks, as discussed, there are other places where the name could be refactored to be consistent across all the code

Comment (JPPortier) [reply client/src/main/com/sinch/sdk/domains/voice/api/v1/ApplicationsService.java]: Done

File: client/src/test/java/com/sinch/sdk/domains/voice/adapters/CalloutsServiceTest.java (hunk 4 above)
Comment (JPPortier) [reply client/src/test/java/com/sinch/sdk/domains/voice/adapters/CalloutsServiceTest.java]: Done

File: client/src/test/java/com/sinch/sdk/domains/voice/api/v1/adapters/VoiceServiceTest.java (hunk 5 above)
Comment (JPPortier) [reply client/src/test/java/com/sinch/sdk/domains/voice/api/v1/adapters/VoiceServiceTest.java:12]: Right done

**Review by asein-sinch — APPROVED** [review]

**Pull request merged**